msm/README.md

```diff
@@ -0,0 +1,51 @@
+## MSM circuit
+
+This is a specialised circuit to compute a large MSM where the elements are
+defined over a foreign field, like the one used by the Pickles verifier.
+For simplicity, we suppose the foreign field is on maximum 256 bits.
+
+The witness is splitted in two different parts:
+- Columns: this is the column of the circuit.
+- Lookups: this contain a list of lookups per row.
+  [MVLookup](https://eprint.iacr.org/2022/1530.pdf) is used. It contains
+  the individual polynomials $f_i(X)$ containing the looked up values, the
+  table/polynomial $t(X)$ the polynomials $f_i(X)$ look into and the
+  multiplicity polynomial $m(X)$ as defined in the paper.
+
+The individual columns of the circuit are defined in `src/column.rs`.
+The polynomials can be expressed using the expression framework, instantiated in `src/column.rs`, with
+```rust
+type E<Fp> = Expr<ConstantExpr<F>, MSMColumn>
+```
+
+The proof structure is defined in `src/proof.rs`. A proof consists of
+commitments to the circuit columns and the lookup polynomials, in addition to
+their evaluations, the evaluation points and the opening proofs.
+The prover/verifier are respectively in `src/prover.rs` and `src/verifier.rs`.
+Note that the prover/verifier implementation is generic enough to be used to
+implement other circuits.
+
+The protocol used is a Plonk-ish arithmetisations without the permutation
```

Comment:
I'm not actually sure that we will use plonkish constraint (https://twitter.com/feministPLT/status/1413815927704014850) at all, but it's not that important.

Reply:
Yeah, we should simply use multi-variate polynomial constraints :)